Source/AwsCommonRuntimeKit/auth/credentials/CRTAWSCredentialsProvider.swift

```diff
@@ -4,51 +4,109 @@
+    ///   - allocator: (Optional) allocator to override
+    /// - Returns: `CredentialsProvider`
+    /// - Throws: CommonRuntimeError.crtError
+    public static func makeDelegate(credentialsProvider: CredentialsProvider,
+                                    shutdownCallback: ShutdownCallback? = nil,
+                                    allocator: Allocator = defaultAllocator) throws -> AwsCredentialsProvider {
+
+        let delegateCore = DelegateCredentialsProviderCore(credentialsProvider)
+        let shutdownCallbackCore = ShutdownCallbackCore({
+            delegateCore.release()
+            shutdownCallback?()
+        })
         let shutdownOptions =  shutdownCallbackCore.getRetainedCredentialProviderShutdownOptions()
         var options = aws_credentials_provider_delegate_options(shutdown_options: shutdownOptions,
                                                                 get_credentials: getCredentialsDelegateFn,
-                                                                delegate_user_data: credProviderPtr)
+                                                                delegate_user_data: delegateCore.passRetained())
 
-        guard let credProvider = aws_credentials_provider_new_delegate(allocator.rawValue, &options) else {
+        guard let provider = aws_credentials_provider_new_delegate(allocator.rawValue, &options) else {
             shutdownCallbackCore.release()
+            delegateCore.release()
             throw CommonRunTimeError.crtError(CRTError.makeFromLastError())
         }
-        self.init(credentialsProvider: credProvider, allocator: allocator)
+        return AwsCredentialsProvider(credentialsProvider: provider,
+                                      allocator: allocator)
     }
 
     /// Creates a credentials provider containing a fixed set of credentials.
     ///
     /// - Parameters:
-    ///   - config:  The `CredentialsProviderStaticConfigOptions` config object.
-    /// - Returns: `AWSCredentialsProvider`
-    public convenience init(fromStatic config: CRTCredentialsProviderStaticConfigOptions,
-                            allocator: Allocator = defaultAllocator) throws {
-
-        let shutdownCallbackCore = ShutdownCallbackCore(config.shutdownCallback)
+    ///   - accessKey: The access key to use.
+    ///   - secret: The secret to use.
+    ///   - sessionToken: (Optional) Session token to use.
+    ///   - shutdownCallback:  (Optional) shutdown callback
+    ///   - allocator: (Optional) allocator to override
+    /// - Returns: `CredentialsProvider`
+    /// - Throws: CommonRuntimeError.crtError
+    public static func makeStatic(accessKey: String,
+                                  secret: String,
+                                  sessionToken: String? = nil,
+                                  shutdownCallback: ShutdownCallback? = nil,
+                                  allocator: Allocator = defaultAllocator) throws -> AwsCredentialsProvider {
```

Comment:
Similar as a comment above, it's preferable to move this back to an initializer rather than a static function